### PR TITLE
Globalize toast and toast when program created

### PR DESCRIPTION
**Pull request by hlminh2000**
**Description of changes**

- Globalizes toast
- Creates an `ApplicationRoot` component to simplify the `_app.js` file
- Toast when program is created

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: client/pages/_app.js [hunk 1]
```diff
@@ -173,4 +101,31 @@ Root.getInitialProps = async ({
...
+    return (
+      <ApplicationRoot egoJwt={egoJwt} apolloCache={apolloCache} pageContext={ctx}>
+        <Component egoJwt={egoJwt} logOut={logOut} pathname={pathname} {...pageProps} />
+      </ApplicationRoot>
+    );
+  };
+  component.getInitialProps = getInitialProps;
+  return component;
+})();
+
```
Comment (hlminh2000) [inline client/pages/_app.js]: having `getInitialProps` below component was reading strange so I just hoisted it up

File: client/components/ApplicationRoot.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
+//@flow
```
Comment (hlminh2000) [inline client/components/ApplicationRoot.js]: this stuff used to be in `_app.js`, it's getting big so I moved it out

File: client/components/pages/submission-system/layout.js [hunk 3]
```diff
@@ -51,65 +51,40 @@ export const ModalPortal = ({ children }: { children: React.Node }) => {
...
+        <Footer />
+      </PageFooter>
+      <div
+        css={css`
+          position: fixed;
+          z-index: 9999;
+        `}
+        ref={modalPortalRef}
+      />
+    </PageContainer>
```
Comment (hlminh2000) [inline client/components/pages/submission-system/layout.js]: moved all the toast stuff out of here into `ApplicationRoot`

Comment (hlminh2000) [reply client/components/pages/submission-system/layout.js]: should do the same to modal too

File: client/components/pages/submission-system/create-program/index.js [hunk 4]
```diff
@@ -1,25 +1,45 @@
...
   ${({ theme }) => css(theme.typography.subtitle2)};
   color: ${({ theme }) => theme.colors.secondary};
 `;
 
-export default ({ logOut, pathname }) => {
+export default () => {
+  const toaster = useToaster();
+  const router = useRouter();
+  const onProgramCreated = ({ programName, shortName: programShortName }) => {
+    toaster.addToast({
```
Comment (joneubank) [inline client/components/pages/submission-system/create-program/index.js]: Do you handle the error case?

Comment (hlminh2000) [reply client/components/pages/submission-system/create-program/index.js]: Great point :) thanks!

_Sent from my OnePlus5 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

File: client/components/pages/submission-system/create-program/index.js [hunk 5]
```diff
@@ -37,6 +37,13 @@ export default () => {
       },
     });
   };
+  const onProgramCreationError = (err: Error) => {
+    toaster.addToast({
+      title: '',
+      variant: TOAST_VARIANTS.ERROR,
+      content: 'Something went wrong, please try again later or contact us for assistance',
```
Comment (joneubank) [inline client/components/pages/submission-system/create-program/index.js]: This is fine for now. We need to standardize our error messages.

**Pull request merged**